File: Python/StoryDownload.py
import requests
import json
import re
import urllib.request
import argparse
ids=lambda user: requests.get('https://www.instagram.com/'+user).text
def download(ids,userat):
    ids=re.search('"profilePage_(.*?)",',ids).group(1)    
    req=requests.get('http://mrvirus.cf/StoryDownload.php?id='+ids).text
    x=re.findall('{URL:(.*?)}',req)
    ib=0
    for i in x:
        if 'mp4' in i:
            urllib.request.urlretrieve(i,userat+str(ib)+'.mp4')
            ib+=1

        else:
            urllib.request.urlretrieve(i,userat+str(ib)+'.jpg')
            ib+=1
    return True
def parse():
    arg=argparse.ArgumentParser()
    arg.add_argument('-u','--user',dest='user',help='-u , --user use to pass user instagram')    
    return arg.parse_args().user
user=parse()
if(user):
    ids=ids(user)
    download(ids,user)
    input('Finished...')
else:
    user=str(input('username : '))
    ids=ids(user)
    download(ids,user)
    input('Finished...')

File: PHP/storydownload.php
<?php
set_time_limit(0);
/////////// Check Request Method
if($_SERVER['REQUEST_METHOD']=='GET' and isset($_GET['user']))
{
	$path= file_exists('Dstory');
	if($path){work($_GET['user']);}
	else{mkdir('Dstory');work($_GET['user']);}
	
		
	
	
	
}

///////// Get ID user To Start Our Mission
function getid($user)
{
	$result=file_get_contents('https://www.instagram.com/'.$user);
	preg_match('/"profilePage_(\d+)",/',$result,$out);
	return $out[1];
}
////// Get URLs From ID OF user 
function fillter($id)
{
	$result=file_get_contents('http://mrvirus.cf/StoryDownload.php?id='.$id);
	preg_match_all('/{URL:(.*?)}/',$result,$out);
	return $out[1];
}
/////// Download Videos And Photos
function download($user,$url,$count)
{
	$result=file_get_contents($url);
	if(stristr($result,'mp4'))
		{
		 $down=fopen('Dstory\\'.$user.$count.'.mp4','wb');
		 fwrite($down, $result);
		 fclose($down);
		 sleep(1);
		}
	else{
		 $down=fopen('Dstory\\'.$user.$count.'.jpg','wb');
		 fwrite($down, $result);
		 fclose($down);
		 sleep(1);
		}
}
//// Work Function To Do our Thins From scratch
function work($user)
{
$ids=getid($user);
$listURL=fillter($ids);

$count =0;
foreach($listURL as $url){
download($user,$url,$count);
$count =$count+1;
}
echo "Finished..";
}

?>